docs/smart-contracts/guides/events/consume.mdx

```diff
@@ -1,7 +1,251 @@
 ---
 title: Consume previously ingested events
-hide_table_of_contents: true
-draft: true
 ---
 
-Placeholder text goes here.
+Once events have been ingested into a database, for instance as done in the [ingest guide], they can be consumed without having the need to query again Soroban RPC. In the following, we will show how we can consume these events in Python.
+
+To access the database, we will use [SQLAlchemy](https://www.sqlalchemy.org), which is one of the most used library used in Python to query database. Similarly to Prisma, SQLAlchemy is an Object Relational Mapper (ORM), allowing us to write database query directly in Python.
+
+Then, we will use [stellar-sdk] to decode the events and process them.
+
+Let's get started!
+
+## Setup
+
+In a virtual environment, install the Python dependencies:
+
+```bash
+pip install sqlalchemy stellar-sdk
+```
+
+## Events in a DB
+
+First, we will need some events in a database. Say we have ingested events in a table named `Event` which has two columns: `topics: dict(str)` and `value: str`. In SQLAlchemy, this translates into a class, also called a database model:
+
+```python
+from typing import Any
+from sqlalchemy import orm, JSON
+
+
+class Base(orm.DeclarativeBase):
+    # needed to tell SQLAlchemy to translate a dictionary into a JSON entry
+    type_annotation_map = {
+        dict[str, Any]: JSON,
+    }
+
+
+class Event(Base):
+    __tablename__ = "event"
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    topics: orm.Mapped[dict[str, Any]]
+    value: orm.Mapped[str]
+```
+
+:::info
+
+Using a database model is very convenient as it allows us to control the database schema programmatically. If we need to change the schema, by adding a new columns for instance, then using an ORM allows to use very powerful migration tools.
+
+:::
+
+We will use an in-memory-only SQLite database for this guide, but thanks to the use of an ORM, we could be using any other supported database. We would simply need to change the connection string.
+
+```python
+from sqlalchemy import create_engine
+engine = create_engine("sqlite://", echo=True)
+
+# the following creates the table in the DB
+Base.metadata.create_all(engine)
+```
+
+By setting `echo=True` we can understand what is happening on the database. Creating the database table leads to the following logs:
```

Comment:
I actually also never really use this and I should 😅 I typically rely on tests or a local DB and since here I did not want to introduce something else, I went with this.